### PR TITLE
Save publisher payouts

### LINKED ISSUE #178: Store publisher payout history
We should store publisher payouts and show these to publishers. Each record needs to have the time frame it was for, the amount, how it was paid out, and maybe a link to a confirmation if applicable. Publishers should be able to browse their past payouts.

### PR DESCRIPTION
**Pull request by davidfischer**
Store publisher payout including the date of the payout, the amount, and any confirmation details, notes, or attachments. ~~Because #222 has a migration, this might need an update after that is merged.~~

Fixes #178

### CODE REVIEW

File: adserver/admin.py [hunk 1]
```diff
@@ -774,7 +775,17 @@ class ViewAdmin(AdBaseAdmin):
     model = View
 
 
+class PublisherPayoutAdmin(admin.ModelAdmin):
+    list_display = ("amount", "publisher", "date", "modified", "created")
```
Comment (ericholscher) [inline adserver/admin.py:779]: Might show the publisher first here? 
```suggestion
    list_display = ("publisher", "amount", "date", "modified", "created")
```

File: adserver/templates/adserver/publisher/payouts.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{% extends "adserver/base.html" %}
```
Comment (ericholscher) [inline adserver/templates/adserver/publisher/payouts.html:1]: Maybe call this template `payout_list.html`? I got confused with just the plural.

File: adserver/templates/adserver/publisher/payouts.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  {{ block.super }}
+  <li class="breadcrumb-item"><a href="{% url 'publisher_main' publisher.slug %}">{{ publisher }}</a></li>
+  <li class="breadcrumb-item active">{% trans 'Payouts' %}</li>
+{% endblock breadcrumbs %}
+
+
+{% block content_container %}
+
+<h1>{% block heading %}{% trans "Publisher payouts" %}{% endblock heading %}</h1>
+
```
Comment (ericholscher) [inline adserver/templates/adserver/publisher/payouts.html:19]: We should probably have some copy here explaining our payout policies, and links to some docs perhaps? Definitely need at least some introductory copy.

File: adserver/views.py [hunk 4]
```diff
@@ -823,6 +825,55 @@ def get_success_url(self):
...
+
+    model = PublisherPayout
+    template_name = "adserver/publisher/payouts.html"
+
+    def get_context_data(self, **kwargs):  # pylint: disable=arguments-differ
+        context = super().get_context_data(**kwargs)
+
+        payouts = self.get_queryset()
+        total = payouts.aggregate(
+            total=models.Sum("amount", output_field=models.FloatField())
```
Comment (ericholscher) [inline adserver/views.py:840]: Seems odd to use a Float to count money, is there not Decimal support here?

File: adserver/admin.py (hunk 1 above)
Comment (davidfischer) [reply adserver/admin.py:779]: My reasoning for not doing that is that makes the publisher name the clickable link but it leads to a payout, not to the publisher. I also considered maybe displaying the PK as the clickable link. Thoughts?

File: adserver/views.py (hunk 4 above)
Comment (davidfischer) [reply adserver/views.py:840]: The underlying field where the payout is stored is an exact decimal field. Only when it is aggregated is it rounded into a float field. I didn't actually test whether a decimal field was supported here.

File: adserver/templates/adserver/publisher/payouts.html (hunk 3 above)
Comment (davidfischer) [reply adserver/templates/adserver/publisher/payouts.html:19]: Where should those docs live? Not in the ad server, I'd think.

File: adserver/views.py (hunk 4 above)
Comment (ericholscher) [reply adserver/views.py:840]: Yea -- I think we need to get the website built out. Having a place for blog posts, policy, etc. seems more and more important.

Comment (davidfischer) [reply adserver/views.py:840]: A decimal field will work fine for aggregating.

File: adserver/admin.py (hunk 1 above)
Comment (ericholscher) [reply adserver/admin.py:779]: I think its probably fine as you have it, for now.

Comment (davidfischer) [reply adserver/admin.py]: I ended up pushing a change that linked the PK.

**Comment by davidfischer** [comment]
I pushed a few changes with some fixes:

- Fix the migration order
- Use decimalfield for aggregating
- payouts -> payout-list
- clickable link for a payout PK

**Pull request merged**